modules/basegl/src/processors/embeddedvolumeslice.cpp

```diff
@@ -0,0 +1,243 @@
+        for (auto& point : intersections) {
+            const vec3 candVector = glm::normalize(point - midpoint);
+            const auto dotProdModifier = glm::dot(candVector, mainNormal);
+            auto dotProd = glm::dot(candVector, mainVector);
+            if (dotProdModifier < 0 && candVector != mainVector) {
+                // let everything past 180 deg be in the domain -3 < x < -1
+                dotProd = -(dotProd + 2);
+            }
+            dotProdVals.push_back(dotProd);
+        }
+
+        std::vector<unsigned int> idx(dotProdVals.size());
+        std::iota(idx.begin(), idx.end(), 0);
+
+        sort(idx.begin(), idx.end(),
+             [&dotProdVals](size_t i1, size_t i2) { return dotProdVals[i1] > dotProdVals[i2]; });
+
+        if (embeddedMesh_.getNumberOfIndicies() == 0) {
+            embeddedMesh_.addIndexBuffer(DrawType::Triangles, ConnectivityType::None);
+        }
+
+        auto& inds =
+            embeddedMesh_.getIndices(0)->getEditableRAMRepresentation()->getDataContainer();
+        inds.clear();
+
+        intersections.push_back(midpoint);
+        for (size_t i = 0; i < idx.size() - 1; i++) {
+            inds.push_back(idx[i]);
+            inds.push_back(idx[i + 1]);
+            inds.push_back(static_cast<unsigned int>(intersections.size() - 1));  // midpoint
+        }
+        inds.push_back(idx[idx.size() - 1]);
+        inds.push_back(idx[0]);                                               // wrap-around
+        inds.push_back(static_cast<unsigned int>(intersections.size() - 1));  // midpoint
+
+        embeddedMesh_.setModelMatrix(inport_.getData()->getModelMatrix());
+        embeddedMesh_.setWorldMatrix(inport_.getData()->getWorldMatrix());
+    }
+}
+
+void EmbeddedVolumeSlice::handlePicking(PickingEvent* p) {
+    if (inport_.hasData() && p->getHoverState() == PickingHoverState::Move &&
+        p->getPressItems().empty()) {
+        const auto data = inport_.getData();
+
+        const Plane plane{*planePosition_, glm::normalize(*planeNormal_)};
+        const auto& ct = data->getCoordinateTransformer(camera_);
+
+        const vec3 ndc1{p->getNDC().x, p->getNDC().y, -1.0};
+        const auto worldPos1 = camera_.getWorldPosFromNormalizedDeviceCoords(ndc1);
+        const auto dataPos1 = vec3{ct.getWorldToDataMatrix() * vec4{worldPos1, 1.0f}};
+
+        const vec3 ndc2{p->getNDC().x, p->getNDC().y, 1.0};
+        const auto worldPos2 = camera_.getWorldPosFromNormalizedDeviceCoords(ndc2);
+        const auto dataPos2 = vec3{ct.getWorldToDataMatrix() * vec4{worldPos2, 1.0f}};
+
+        if (const auto intersection = plane.getIntersection(dataPos1, dataPos2);
+            intersection.intersects_) {
+            const auto dataPoint = intersection.intersection_;
+            const auto index =
+                static_cast<size3_t>(vec3{ct.getDataToIndexMatrix() * vec4{dataPoint, 1.0f}});
+
+            const auto cind = glm::clamp(index, size3_t{0}, data->getDimensions() - size3_t{1});
```

Comment:
Is it reasonable to clamp the index instead of just discarding it?

Reply:
Should not happen, but I want to avoid any potential crash....